### ISSUE TITLE
Store user's Offense/Defense/Spy/Sentry stats in DB

### ISSUE THREAD

**Issue by uaktags** [OpenThrone/OpenThrone#192]
While looking into #171 it's obvious that the calculations should never have been tied to the user model, which gets counted dynamically every time it's referenced, and should be calculated and stored in the DB.

To do that, we'll need to update the following events:
- When an attack happens and you lose units
- When you train/untrain
- when you equip/unequip
- BattleUpgrades/StructureUpgrades
- Any others? (we should be using the services though to manage it centrally)

While we're at it, we probably should be calculating and storing Killing Strength and Defense Strength as well. This may take us a little bit and will surely cause an update to Prisma Schema

**Comment by uaktags** [comment]
@MarkTraceur Another JSON blob :-P 

Do you want me to tie this with the later hopes for replacing json blobs as a whole or continue without it? I can reuse the current JSON blob of user.stats (semi serious, more so joking...but yes?) or we can just start off with a schema change:
```
killing_str        Int       @default(1)
defense_str        Int       @default(1)
spying_str         Int       @default(1)
sentry_str         Int       @default(1)
  ```
  
  The only issue i have with DB storage is that we currently use KS/DS calculations on a per-turn basis along with applying deaths on a per turn basis (see #191 discussion), so we'd would need to still calculate on the fly. But this would atleast be on the serverside and just during the attacking calculation, speed should be optimized for all other actions.

**Comment by MarkTraceur** [comment]
Replace one thing first, then replace the other thing later if needed :) don't let perfect be the enemy of good!

**Comment by uaktags** [comment]
closed with #197
